pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -234,7 +246,17 @@ func (c *Controller) isBeingUsed(pvc *v1.PersistentVolumeClaim) (bool, error) {
 	// mean such a Pod doesn't exist: it might just not be in the cache yet. To
 	// be 100% confident that it is safe to delete pvc make sure no Pod is using
 	// it among those returned by a live list.
-	return c.askAPIServer(pvc)
+	inUse, err := c.askAPIServer(pvc)
+	if err != nil {
+		return false, err
```

Comment:
I see the possible error is pods are failed to list. To be safer, I think we should not return false.

Reply:
Didn't quite understand the comment. If we return an error != nil, the bool value is ignored and function returns 
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/pvcprotection/pvc_protection_controller.go#L175

Reply:
Here I mean you can continue to check volumeInUse instead of return false? False means pvc is not being used, and finalizer will be removed, right?

Reply:
No since err != nil, finalizer will not be removed
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/pvcprotection/pvc_protection_controller.go#L176

Reply:
got it. I am also thinking it is better to check volumeInUse first. The number of nodes should be much smaller than number of pods. Also askAPIServer is calling API server directly instead of using informer cache?

Reply:
Ack. Changed the order of the check : nodes checked first before pods.